crates/evm/src/either.rs

```diff
@@ -0,0 +1,126 @@
+//! Helper type that represents one of two possible executor types
+
+use crate::execute::{
+    BatchBlockExecutionOutput, BatchExecutor, BlockExecutionInput, BlockExecutionOutput,
+    BlockExecutorProvider, Executor,
+};
+use reth_interfaces::{executor::BlockExecutionError, provider::ProviderError};
+use reth_primitives::{BlockNumber, BlockWithSenders, PruneModes, Receipt};
+use revm_primitives::db::Database;
+
+/// A type that represents one of two possible executor factories.
+#[derive(Debug, Clone)]
+pub enum EitherExecutor<A, B> {
```

Comment:
why not either::Either?

Reply:
good point, will just import futures::Either